node/engine/messageservice/p2p-message-service/service.go

```diff
@@ -132,6 +134,9 @@ func (ms *P2PMessageService) setupMdns() {
 	ms.mdns = mdns.NewMdnsService(ms.p2pHost, "", ms)
 	err := ms.mdns.Start()
 	ms.checkError(err)
+
+	ms.InitComplete <- true
```

Comment:
Did you consider simply closing the `chan` instead?

Reply:
I did not consider it but thats probably a better implementation in case there are multiple listeners on that channel. I will make that change

Reply:
Done in https://github.com/statechannels/go-nitro/pull/1515/commits/ecfd7dfb5ea14f594288e01681f05f3e50c71f76